pandas/core/reshape/reshape.py

```diff
@@ -689,7 +689,7 @@ def _convert_level_number(level_num, columns):
         new_labels = [np.arange(N).repeat(levsize)]
         new_names = [this.index.name]  # something better?
 
-    new_levels.append(level_vals)
+    new_levels.append(frame.columns.levels[level_num])
```

Comment:
Since it is in your commit, you somehow made that change :-) 
But if it was not the intent, you can just revert it (change it back to how it was based on the diff you see here)